### PR TITLE
Make tags toolbar and modal like inventory

### PR DESCRIPTION
**Pull request by LI0131**
This PR adds a couple of different features both of which are intended to make the advisor tags filtering system look and feel more like inventory. Namely we are updating the TagsToolbar Select component and swapping out the ManageTags modal for the fec TagModal.

These changes are covered in the following tickets:
https://projects.engineering.redhat.com/browse/RHCLOUD-6740
https://projects.engineering.redhat.com/browse/RHCLOUD-6743
https://projects.engineering.redhat.com/browse/RHCLOUD-6738

Below shows what the changes will look like in Advisor 😁

Screen Shot 2020-07-01 at 2 45 18 PM

Screen Shot 2020-07-01 at 2 45 30 PM

### CODE REVIEW

**Comment by LI0131** [comment]
As a sidenote...

There are a few things that we will still need to implement after this goes in. I'll list them here so we don't 🤦

1. Update TagModal to support `primaryToolbarProps`. This will allow us to pass an `activeFilterConfig` object to the PrimaryToolbar component, which will manage our Filter Chips
2. Update the Title of the TagModal to reflect the mocks (https://marvelapp.com/4e20f90/screen/69662590)
3. Update the /tags API to make the pagination actually work

**Comment by mhuth** [comment]
Is it possible/difficult to group the selections under a common heading? Eg

Satellite
- Capsules
- Ceph3
- Drones
- Host Collection 1
....

insights-client
- key1
- key2
- key3
... etc

**Comment by LI0131** [comment]
@mhuth That's way that inventory is currently displaying tags in their Select component.

Screen Shot 2020-07-02 at 9 03 10 AM

I went ahead and did it this way because that's the way the mocks had it 🤷 but what're your thoughts @AllenBW ?

**Comment by AllenBW** [comment]
💯 ignore the mocks n go with the common heading, might make my first quibble null, if we can fit 20 in a reasonable space lets do it!

**Comment by LI0131** [comment]
The tags should now be grouped by `namespace` 😁 

Screen Shot 2020-07-02 at 10 55 32 AM

File: src/Messages.js [hunk 1]
```diff
@@ -257,6 +257,31 @@ export default defineMessages({
         description: 'Filter text input, for tagstoolbar',
         defaultMessage: 'Filter by tags'
     },
+    filterTagsInModal: {
+        id: 'filterTagsInModal',
+        description: 'Filter text input, for ManageTags Modal',
+        defaultMessage: 'Filter tags'
+    },
+    tagsModalTitle: {
```
Comment (AllenBW) [inline src/Messages.js:265]: this is already defined on line 849, lets just `.toLowerCase()` that value

File: src/PresentationalComponents/TagsToolbar/ManageTags.js [hunk 2]
```diff
@@ -1,36 +1,113 @@
-import { Modal } from '@patternfly/react-core/dist/js/components/Modal/Modal';
+import React, { useEffect, useState } from 'react';
+
+import { DEBOUNCE_DELAY } from '../../AppConstants';
 import PropTypes from 'prop-types';
-import React from 'react';
-import TagsList from './TagsList';
+import { TagModal } from '@redhat-cloud-services/frontend-components/components/TagModal';
 import { connect } from 'react-redux';
+import debounce from '../../Utilities/Debounce';
```
Comment (AllenBW) [inline src/PresentationalComponents/TagsToolbar/ManageTags.js:7]: excellent 😁

File: src/PresentationalComponents/TagsToolbar/ManageTags.js [hunk 3]
```diff
@@ -1,36 +1,113 @@
...
+            loaded,
+            pagination: {
+                perPage,
+                page,
+                count: totalTags
+            },
+            rows: tags.map(tag => ({
+                id: tag.id,
+                selected: selected.find(selection => selection.id === tag.id),
+                cells: [tag.key, tag.value, tag.namespace]
```
Comment (AllenBW) [inline src/PresentationalComponents/TagsToolbar/ManageTags.js:54]: is the system count data available here?  i know its not in the mocks.. but we do display it in the toolbar, kinda seems silly the table doesn't also display it 🤔

File: src/PresentationalComponents/TagsToolbar/ManageTags.js [hunk 4]
```diff
@@ -1,36 +1,113 @@
...
+import React, { useEffect, useState } from 'react';
+
+import { DEBOUNCE_DELAY } from '../../AppConstants';
 import PropTypes from 'prop-types';
-import React from 'react';
-import TagsList from './TagsList';
+import { TagModal } from '@redhat-cloud-services/frontend-components/components/TagModal';
 import { connect } from 'react-redux';
+import debounce from '../../Utilities/Debounce';
 import { injectIntl } from 'react-intl';
```
Comment (AllenBW) [inline src/PresentationalComponents/TagsToolbar/ManageTags.js:8]: i know i know, not related to the work.. but lets use `import { useIntl } from 'react-intl';` here and not pass it through the props (means we can cleanup it on the props).... checkout  src/export/systemspdf.js for an exmaple

File: src/PresentationalComponents/TagsToolbar/ManageTags.js [hunk 5]
```diff
@@ -1,36 +1,113 @@
...
+    totalTags: PropTypes.number.isRequired,
+    isOpen: PropTypes.bool.isRequired,
+    selectedTags: PropTypes.array,
+    setSelectedTags: PropTypes.func,
+    intl: PropTypes.any
+};
 
-export default injectIntl(connect(mapStateToProps)(ViewHostAcks));
+export default injectIntl(connect(
+    state => ({ selectedTags: state.AdvisorStore.selectedTags }),
```
Comment (AllenBW) [inline src/PresentationalComponents/TagsToolbar/ManageTags.js:110]: OOO OOOO deconstruct Advisorstore!

```suggestion
    ({AdvisorStore}) => ({ selectedTags: AdvisorStore.selectedTags }),
```

File: src/PresentationalComponents/TagsToolbar/TagsToolbar.js [hunk 6]
```diff
@@ -27,34 +28,69 @@ import { useIntl } from 'react-intl';
...
+    const [totalTags, setTotalTags] = useState();
     const [tags, setTags] = useState([]);
     const [searchText, setSearchText] = useState('');
+    const [params, setParams] = useState();
     const debouncedSearchText = debounce(searchText, DEBOUNCE_DELAY);
     const [manageTagsModalOpen, setManageTagsModalOpen] = useState(false);
     const showMoreCount = 20;
 
-    const fetchTags = async (params, filter) => {
+    const groupTags = (tags) => {
```
Comment (AllenBW) [inline src/PresentationalComponents/TagsToolbar/TagsToolbar.js:39]: :godmode:

File: src/PresentationalComponents/TagsToolbar/TagsToolbar.js [hunk 7]
```diff
@@ -104,24 +141,37 @@ const TagsToolbar = ({ selectedTags, setSelectedTags }) => {
...
+                        </span>
+                    )}
+                </SelectGroup>
             )}
             <Button key='manage all tags'
-                variant='link' onClick={(toggleModal) => setManageTagsModalOpen(toggleModal)}>
-                {(showMoreCount > 0 && tags.length > showMoreCount) ?
-                    intl.formatMessage(messages.countMoreTags, { count: tags.length - showMoreCount }) : intl.formatMessage(messages.manageTags)
+                variant='link' onClick={() => setManageTagsModalOpen(true)}>
+                {(showMoreCount > 0 && tags.flatMap(source => source.data).length >= showMoreCount) ?
```
Comment (AllenBW) [inline src/PresentationalComponents/TagsToolbar/TagsToolbar.js:170]: `tags.flatMap(source => source.data).length` happens twice, any way we can make it happen once?

File: src/PresentationalComponents/TagsToolbar/TagsToolbar.js [hunk 8]
```diff
@@ -27,34 +28,74 @@ import { useIntl } from 'react-intl';
...
                 setSelectedTags(tagsToSet);
             }
         } catch (error) {
             addNotification({ variant: 'danger', dismissable: true, title: intl.formatMessage(messages.error), description: `${error}` });
         }
+
+        return formattedTags;
     };
 
+    useEffect(() => {
```
Comment (AllenBW) [inline src/PresentationalComponents/TagsToolbar/TagsToolbar.js:83]: noice

**Pull request merged**